pepmatch/matcher.py

```diff
@@ -838,21 +836,31 @@ def filter_fragments(group: pd.DataFrame) -> pd.DataFrame:
 
     return df
 
+  def _get_function_pointer(self, df: pd.DataFrame):
+    """Returns the df conversion function, given the required output format
+    
+    Args:
+      df: the dataframe of the matches."""
+    return [df.to_csv, df.to_csv, df.to_excel, df.to_json, df.to_html][
+      ['csv', 'tsv', 'xlsx', 'json', 'html'].index(self.output_format)
+    ]
 
   def _output_matches(self, df: pd.DataFrame) -> None:
     """Write Pandas dataframe to format that is specified
-    
+
     Args:
       df: the dataframe of the matches."""
     
-    path = f'{self.preprocessed_files_path}/{self.output_name}'
-    if self.output_format == 'csv':
-      return df.to_csv(f'{path}.csv', index=False)
-    elif self.output_format == 'tsv':
-      return df.to_csv(f'{path}.tsv', sep='\t', index=False)
-    elif self.output_format == 'xlsx':
-      return df.to_excel(f'{path}.xlsx', index=False)
+    # appends '.' + filetype if the name does not already contain it
+    path = self.output_name.__str__()
+    if path.split('.')[-1] != self.output_format:
+      path += f".{self.output_format}"
+
+    output = self._get_function_pointer(df)
```

Comment:
can we call `_get_function_pointer` `_get_output_function` instead?

Reply:
You bet